src/overlays/actors/ovl_Obj_Warp2block/z_obj_warp2block.c

```diff
@@ -43,36 +42,305 @@ const ActorInit Obj_Warp2block_InitVars = {
+    f32 temp_f0;
+    Actor* temp_a3;
+    Player* player;
+    Vec3f sp20;
+    f32 temp_f2;
+
+    if (func_80043590(&this->dyna) != 0) {
+        return 0;
+    }
+
+    temp_a3 = this->dyna.actor.child;
+	player = PLAYER;
+    if ((this->dyna.actor.xzDistToPlayer <= sDistances[(((this->dyna.actor.params >> 0xB) & 7))]) || 
+    	(temp_a3->xzDistToPlayer <= sDistances[(((temp_a3->params >> 0xB) & 7))])) {
+
+        func_8002DBD0(&this->dyna.actor, &sp20, &player->actor.world.pos);
+        temp_f2 = (this->dyna.actor.scale.x * 50.0f) + 6.0f;
+
+        if (!(temp_f2 < fabsf(sp20.x)) && !(temp_f2 < fabsf(sp20.z))) {
+            return 0;
+        }
+
+        func_8002DBD0(temp_a3, &sp20, &player->actor.world.pos);
+        temp_f2 = (temp_a3->scale.x * 50.0f) + 6.0f;
+
+        if (!(temp_f2 < fabsf(sp20.x))) {
+            temp_f0 = fabsf(sp20.z);
+            if (!(temp_f2 < temp_f0)) {
+                return 0;
+            }
+        }
+    } else {
+    	return 0;
+    }
+
+    return 1;
+}
+
+
+
+void ObjWarp2block_SwapWithChild(ObjWarp2block *this, GlobalContext *globalCtx) {
+    Vec3f tempVec;
+    Vec3s tempRot;
+    s32 temp;
+
+    Math_Vec3f_Copy(&tempVec, &this->dyna.actor.world.pos);
+    Math_Vec3f_Copy(&this->dyna.actor.world.pos, &this->dyna.actor.child->world.pos);
+    Math_Vec3f_Copy(&this->dyna.actor.child->world.pos, &tempVec);
+
+    temp = this->dyna.actor.world.rot.y;
+    this->dyna.actor.world.rot.y = this->dyna.actor.child->world.rot.y;
+    this->dyna.actor.child->world.rot.y = temp;
+
+    temp = this->dyna.actor.shape.rot.y;
+    this->dyna.actor.shape.rot.y = this->dyna.actor.child->shape.rot.y;
+    this->dyna.actor.child->shape.rot.y = temp;
+
+    temp = this->dyna.actor.home.rot.z;
+    this->dyna.actor.home.rot.z = this->dyna.actor.child->home.rot.z;
+    this->dyna.actor.child->home.rot.z = temp;
+
+    Math_Vec3f_Copy(&tempVec, &this->dyna.actor.scale);
+    Math_Vec3f_Copy(&this->dyna.actor.scale, &this->dyna.actor.child->scale);
+    Math_Vec3f_Copy(&this->dyna.actor.child->scale, &tempVec);
+
+    Math_Vec3f_Copy(&tempVec, &this->dyna.actor.focus.pos);
+    Math_Vec3f_Copy(&this->dyna.actor.focus.pos, &this->dyna.actor.child->focus.pos);
+    Math_Vec3f_Copy(&this->dyna.actor.child->focus.pos, &tempVec);
+
+    tempRot = this->dyna.actor.focus.rot;
+    this->dyna.actor.focus.rot = this->dyna.actor.child->focus.rot;
+    this->dyna.actor.child->focus.rot = tempRot;
+
+    temp = this->dyna.actor.params & 0x7FFF;
+    this->dyna.actor.params = (this->dyna.actor.params & 0x8000) | (this->dyna.actor.child->params & 0x7FFF);
+    this->dyna.actor.child->params = (this->dyna.actor.child->params & 0x8000) | (temp  & 0x7FFF);
+
+    if (Math3D_Vec3fDistSq(&this->dyna.actor.world.pos, &this->dyna.actor.home.pos) < 0.01f) {
+        Flags_UnsetSwitch(globalCtx, this->dyna.actor.params & 0x3F);
+    } else {
+    	Flags_SetSwitch(globalCtx, this->dyna.actor.params & 0x3F);
```

Comment:
```suggestion
    	  Flags_SetSwitch(globalCtx, this->dyna.actor.params & 0x3F);
```

Reply:
Sorry for the myriad of small spacing inconsistencies, I was on a new VM and my editor wasn't being consistent with tabs and spaces. Should be fixed everywhere now.